comp/core/telemetry/component_mock.go

```diff
@@ -12,10 +12,19 @@ import (
 	sdk "go.opentelemetry.io/otel/sdk/metric"
 )
 
+// Metrcis interface defines the retrival functions to extract information from a metric
+type Metric interface {
+	Labels() map[string]string
```

Comment:
Prometheus uses "labels" but I agree "tags" is the common term at Datadog, and it would be more explicit
(and here using Prometheus should be an implementation detail anyway)

Reply:
The internal of prometheus calls it Labels, and we retrieve the values using `GetLabel()`, so I think keeping the name label makes sense here. We could rename the function to `GetLabels` and the `GetValue()`.

What do you think?